### ISSUE TITLE
[BUG] alphabetize member-ordering-extended shows wrong lints

### ISSUE THREAD

**Issue by grafovdenis** [dart-code-checker/dart-code-metrics#345]
* **Dart code metrics version:** 3.2.2
* **Dart sdk version:** 2.12.0

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  rules:
    - member-ordering-extended:
        alphabetize: true
        order:
          - constructors
          - named-constructors
          - overridden-methods
          - public-methods
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
class OrderingExample {
  const OrderingExample();

  @override
  String toString() {
    return super.toString().substring(0);
  }

  void foo() { // (1) => foo should be alphabetically before toString.
    throw UnimplementedError();
  }
}

class Ordering {
  void foo() {
    throw UnimplementedError();
  }

  const Ordering(); // (2) => constructors should be before overridden-methods.

  @override
  String toString() {
    return super.toString().substring(0);
  }
}
```

**What did you expect to happen?**

1. Another lint: `public-methods should be before overridden-methods.`
2. Lint should apear above `foo` function: `public-methods should be before constructors`

**What actually happened?**

Look at comments in code.

**Comment by grafovdenis** [comment]
Found few more bugs:
**Config (order only):**
```yaml
          - overridden-methods
          - public-static-methods
          - public-methods
          - protected-methods
          - private-static-methods
          - private-methods
```

```dart
class Example {
  @override
  String toString() {
    return super.toString().substring(0);
  }

  static void staticFoo() {
    return;
  }

  void foo() { // (3) => overridden-methods should be before public-static-methods.
    return;
  }

  @protected // (4) => overridden-methods should be before public-static-methods.
  void protectedFoo() {
    return;
  }

  // ignore: unused_element
  static void _staticFoo() {
    return;
  }

  // ignore: unused_element
  void _foo() { // (5) => overridden-methods should be before private-static-methods.
    return;
  }
}
```

**Comment by incendial** [comment]
@grafovdenis I've finally looked closer and fix is on the way. 

> 2. Lint should appear above foo function: public-methods should be before constructors

Are you sure that this is the expected result? From my point of view it should be `constructors should be before public-methods.`.

> Found few more bugs:

Am I correct that the expected result was to have 0 issues?

**Comment by grafovdenis** [comment]
@incendial Nicely done!
> Are you sure that this is the expected result? From my point of view it should be `constructors should be before public-methods.`.

Yep, you are right.

> Am I correct that the expected result was to have 0 issues?

Yes, there should be no issues.

**Comment by incendial** [comment]
3.3.0 version with the fix is now available on pub.
